package/MDAnalysis/analysis/base.py

```diff
@@ -125,14 +125,105 @@
 import inspect
 import logging
 import itertools
+import warnings
 
 import numpy as np
 from MDAnalysis import coordinates
 from MDAnalysis.core.groups import AtomGroup
 from MDAnalysis.lib.log import ProgressBar
+from typing import Callable, Iterable
+from importlib import import_module
+from functools import partial
 
 logger = logging.getLogger(__name__)
 
```

Comment:
Here and everywhere else - please have a look & address the many flake8 found in tthe diff (https://github.com/MDAnalysis/mdanalysis/actions/runs/5683556166/job/15404355400)

Please ignore all the `black` stuff, just the lines that look like this: https://github.com/MDAnalysis/mdanalysis/actions/runs/5683556166/job/15404355400#step:4:617

Reply:
I ran `black` on `base.py` since it's the file I edit the most, but I'm yet hesitant to run it on other files that I'm working on, like `rms.py` in this PR for example.

Reply:
Please don't run `black` on the whole file. it'll not only make review hard but it'll also make it hard to know that you changed here

Reply:
Unresolving and tagging @RMeli here.

Reply:
See discussion on Discord: https://discord.com/channels/807348386012987462/1105792867143798835/1136030236308418671